### ISSUE TITLE
Crash when starting application and trying to patch the executable

### ISSUE THREAD

**Issue by Pablossjui** [nipkownix/re4_tweaks#52]
When starting RE4_tweaks with www.re4hd.com's mod, and starting the game I get the message that it wants to patch it to support more than 2GB of RAM.

However the screen almost instantly freezes and I cannot select the option to do it. Tried moving mouse, hitting the Enter key or using the arrows and cannot get any response; I have to force open task manager and force close the application in order to free up my system.

https://ntcore.com/?page_id=371 This manual patcher instantly worked and was able to play using the mod right away

AMD Ryzen 5 2600
16.0 GB RAM
GTX 1080 TI 
16 GIG RAM
NO other mods
BIO4 37.4 GB, 6804 files
BIN32 3.92 MB, 4 files
No issues before install

**Comment by nipkownix** [comment]
>https://ntcore.com/?page_id=371 This manual patcher instantly worked and was able to play using the mod right away

After doing this manual patch, does the game freeze if you press F1 to open the configuration menu?

**Comment by nipkownix** [comment]
Interesting. I was suspecting the way I handle input in the menus was the cause of the freezing, but apparently not.

@emoose any idea?

**Comment by laymonage** [comment]
Seems that it's caused by GeForce Experience Overlay:

https://steamcommunity.com/app/254700/discussions/0/3108018050534635927/?ctp=8#c3182358518942093219

**Comment by Saidana94** [comment]
Hello all, 
I think this problem also occurs to those who have a 120hz or 144hz monitor, I have personally installed the mod but I can't access the game. The screen freezes instantly but I can hear the game. This means that I can't even apply the **Fix display mode/black screen issue** mod. 
I will probably have to install the 4GB patch manually and see how it goes. 
This is just my theory, I am not 100% sure !
Cheers. 
PS : I don't have Geforce Experienc Overlay

**Comment by Pablossjui** [comment]
> just to confirm, it doesn't crash? If it did, we could get some log files to investigate.

It'd be more accurate to say it freezes in place and doesn't allow to do anything but you can still see the message on the screen telling you to patch it

I do have the Geforce Experience overlay so that might do it, why though? I have no idea. Could it be because it was in exclusive-fullscreen?

**Comment by nipkownix** [comment]
>It'd be more accurate to say it freezes in place and doesn't allow to do anything but you can still see the message on the screen telling you to patch it

I see.
The game behind the message is frozen as well? Like, can you see any movement or is the image static?

**Comment by nipkownix** [comment]
Just to rule it out, I've changed the way input is handled in the 4gb patch window.

@Pablossjui, could you see if the issue still happens with this build?

[dinput8.zip](https://github.com/nipkownix/re4_tweaks/files/7985664/dinput8.zip)

**Comment by Pablossjui** [comment]
> The game behind the message is frozen as well? Like, can you see any movement or is the image static?

Yes, the background is just black no movement at all.


I did some tests. Thankfully the manual patcher makes a backup of your exe file so I didn't have to redownload the whole game lol

First, I disabled the Nvidia game overlay, and can confirm it worked, it did freeze but for like 2 seconds, but then the game on the background appears and the mouse frees up too, so I can go ahead and patch it using the message.

Then, I enabled the overlay again and the issue returned. With the overlay like that, I replaced the dinput dll with the one you provided and can confirm it also worked; the game again freezed for like 2 seconds, but then it all works out like before.

Finally I AGAIN changed the dinput dll with the one from re4hd's mod and the issue reappeared.


I haven't done any more tests regarding gameplay or stability but it seems to work now with that new dll

**Comment by nipkownix** [comment]
I see, I see. Then it really is the way I handle input that is causing the issue. The weird part is that the configuration menu uses the exact same method, but doesn't cause problems.

Thanks for testing!

**Comment by nipkownix** [comment]
(posting this here since this is about the LAApatch)

@emoose I made some changes to the LAApatch. Now it doesn't use raw input (to not break whatever the Geforce Experience Overlay is doing on startup) and the OK button now forces the game to restart.

Do you think forcing the game to restart is too _drastic_? Unrelated to this issue, but I've seen some people pressing yes on the patch but not actually restarting the game to make the changes work.

**Comment by emoose** [comment]
Oh nice, wanted to add something like that too but wasn't really sure how to go about it, I think it's fine to force it since like you said many people seem to be ignoring us asking to restart.

Guess some don't bother actually reading the prompt before clicking it - maybe should add something in all-caps or bold text that pressing yes will force a restart, since some might be surprised by it.

**Comment by nipkownix** [comment]
> (I wonder if maybe we should be doing the LAA stuff with windows MessageBoxes instead, before game window actually opens up, would probably look cleaner instead of game needing to start up -> exit -> start again)

True. We can keep it the way it is for now, but if some other issue happens like the nvidia one, we just switch to MessageBoxes instead.
